integration-tests/crib/connect.go

```diff
@@ -0,0 +1,120 @@
+package crib
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/seth"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/ptr"
+	seth_utils "github.com/smartcontractkit/chainlink-testing-framework/utils/seth"
+	tc "github.com/smartcontractkit/chainlink/integration-tests/testconfig"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	msClient "github.com/smartcontractkit/chainlink-testing-framework/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+)
+
+const (
+	// these are constants for simulated CRIB that should never change
+	// CRIB: https://github.com/smartcontractkit/crib/tree/main/core
+	// Core Chart: https://github.com/smartcontractkit/infra-charts/tree/main/chainlink-cluster
+	mockserverCRIBTemplate        = "https://%s-mockserver%s"
+	internalNodeDNSTemplate       = "app-node%d"
+	ingressNetworkWSURLTemplate   = "wss://%s-geth-1337-ws%s"
+	ingressNetworkHTTPURLTemplate = "https://%s-geth-1337-http%s"
+)
+
+func setSethConfig(cfg tc.TestConfig, netWSURL string, netHTTPURL string) {
+	netName := "CRIB_SIMULATED"
+	cfg.Network.SelectedNetworks = []string{netName}
+	cfg.Network.RpcHttpUrls = map[string][]string{}
+	cfg.Network.RpcHttpUrls[netName] = []string{netHTTPURL}
+	cfg.Network.RpcWsUrls = map[string][]string{}
+	cfg.Network.RpcWsUrls[netName] = []string{netWSURL}
+	cfg.Seth.EphemeralAddrs = ptr.Ptr(int64(0))
+}
+
+// ConnectRemote connects to a local environment, see https://github.com/smartcontractkit/crib/tree/main/core
+// connects to default CRIB network if simulated = true
+func ConnectRemote(simulated bool) (
+	*seth.Client,
+	*msClient.MockserverClient,
+	*client.ChainlinkK8sClient,
+	[]*client.ChainlinkK8sClient,
+	error,
+) {
+	ingressSuffix := os.Getenv("K8S_STAGING_INGRESS_SUFFIX")
+	if ingressSuffix == "" {
+		return nil, nil, nil, nil, errors.New("K8S_STAGING_INGRESS_SUFFIX must be set to connect to k8s ingresses")
+	}
+	config, err := tc.GetConfig([]string{"CRIB"}, tc.OCR)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	if config.CRIB.CLNodesNum < 2 {
+		return nil, nil, nil, nil, fmt.Errorf("not enough chainlink nodes, need at least 2, TOML key: [CRIB.nodes]")
+	}
+	cfg := config.CRIB
+	mockserverURL := fmt.Sprintf(mockserverCRIBTemplate, cfg.Namespace, ingressSuffix)
+	var sethClient *seth.Client
+	if simulated {
+		netWSURL := fmt.Sprintf(ingressNetworkWSURLTemplate, cfg.Namespace, ingressSuffix)
+		netHTTPURL := fmt.Sprintf(ingressNetworkHTTPURLTemplate, cfg.Namespace, ingressSuffix)
+		setSethConfig(config, netWSURL, netHTTPURL)
+		net := blockchain.EVMNetwork{
+			Name:                 cfg.NetworkName,
+			Simulated:            true,
+			SupportsEIP1559:      true,
+			ClientImplementation: blockchain.EthereumClientImplementation,
+			ChainID:              1337,
```

Comment:
should we support 2337 chain as well? we launch CRIB with 2 chains, no?

Reply:
Not sure we need 2 chains for any tests in `core` right now